### ISSUE TITLE
Tab order is not correct on workspace restore

### ISSUE THREAD

**Issue by pravesh-sharma** [pgadmin-org/pgadmin4#8901]
**_Please note that security bugs or issues should be reported to security@pgadmin.org._**

**Describe the bug**

Tab order is not correct on workspace restore

**To Reproduce**

1. Open 2-3 query tool tabs
2. Close pgAdmin
3. Reopen pgAdmin

**Expected behavior**

Tab order on reopening should be same when we close the application.

**Comment by anilsahoo20** [comment]
Issues found while preliminary testing
1. For View/Edit Data restore, even if there is data, it’s showing error notifier
2. For title change in any tool, it’s not restoring the name
3. Sometimes after Query tool gets restored, tabs are not getting closed
4. After restoring query tool, if user cancel the `connect server` dialog and clicks on execute button, `connection warning` dialog/reconnect is not working
5. Open multiple empty ERD tool>Click Generate SQL>close app> reopen app> the QT open using generate SQL is not editable
6. PSQL tool in PSQL workspace viewport/UI height is not visible but for default workspace it works fine.

**Comment by casparsch** [comment]
This issue focuses mainly on tabs being re-opened. It is also applicable for windows being opened.
Not sure if it matters but it could/should be a test-case.

**Comment by anilsahoo20** [comment]
1. File open in QT > deleted before you restore - giving 500 server error
2. Open a QT> tick open in a new tab, auto rollback on error, use server cursor, tick all/any Execute settings> close app>open again> the ticked settings are unticked
3. Open QT> Add new connections through the dropdown> write some query or not> on restore of app it shows the old connection not the new connection with the query if written
4. ERD tool not restoring the settings like Notations selected other than default one
5. Open ERD tool for any object/ empty erd tool and create table object> close app>reopen> enter password if not saved>click generate sql>gettting blank QT screen & QT is not closing
6. File open in QT parallel file edited and saved externally- OUTPUT: opens the edited file on restore but it should ask to restore the file or not.
7. On ERD Tool> if the tab is dirty it shows the save icon enabled. but after restore the dirty icon(*) is not coming beside the tab title and save icon is disabled.

**Comment by anilsahoo20** [comment]
Tested and verified on candidate build: https://developer.pgadmin.org/builds/2025-07-23-1/
Package: pip wheel
Environment: macOs Sonoma 14.7.1
